File: api/enums/period.py
import graphene


class PeriodEnums(graphene.Enum):
    JANUARY = "Jan"
    FEBRUARY = "Feb"
    MARCH = "Mar"
    APRIL = "Apr"
    MAY = "May"
    JUNE = "June"
    JULY = "July"
    AUGUST = "Aug"
    SEPTEMBER = "Sept"
    OCTOBER = "Oct"
    NOVEMBER = "Nov"
    DECEMBER = "Dec"
    LAST30DAYS = "last30days"

    @property
    def description(self):
        if self == PeriodEnums.JANUARY:
            return "The month of January"
        elif self == PeriodEnums.FEBRUARY:
            return "The month of February"
        elif self == PeriodEnums.MARCH:
            return "The month of March"
        elif self == PeriodEnums.APRIL:
            return "The month of April"
        elif self == PeriodEnums.MAY:
            return "The month of May"
        elif self == PeriodEnums.JUNE:
            return "The month of June"
        elif self == PeriodEnums.JULY:
            return "The month of July"
        elif self == PeriodEnums.AUGUST:
            return "The month of August"
        elif self == PeriodEnums.SEPTEMBER:
            return "The month of September"
        elif self == PeriodEnums.OCTOBER:
            return "The month of October"
        elif self == PeriodEnums.NOVEMBER:
            return "The month of November"
        elif self == PeriodEnums.DECEMBER:
            return "The month of December"
        elif self == PeriodEnums.LAST30DAYS:
            return "The last 30 days"
        else:
            return "Error, this enum value does not exist."

File: api/src/enums/period.js
import { GraphQLEnumType } from 'graphql'

export const PeriodEnums = new GraphQLEnumType({
  name: 'PeriodEnums',
  values: {
    JANUARY: {
      value: 'january',
      description: 'The month of January.',
    },
    FEBRUARY: {
      value: 'february',
      description: 'The month of February.',
    },
    MARCH: {
      value: 'march',
      description: 'The month of March.',
    },
    APRIL: {
      value: 'april',
      description: 'The month of April.',
    },
    MAY: {
      value: 'may',
      description: 'The month of May.',
    },
    JUNE: {
      value: 'june',
      description: 'The month of June.',
    },
    JULY: {
      value: 'july',
      description: 'The month of July.',
    },
    AUGUST: {
      value: 'august',
      description: 'The month of August.',
    },
    SEPTEMBER: {
      value: 'september',
      description: 'The month of September.',
    },
    OCTOBER: {
      value: 'october',
      description: 'The month of October.',
    },
    NOVEMBER: {
      value: 'november',
      description: 'The month of November.',
    },
    DECEMBER: {
      value: 'december',
      description: 'The month of December.',
    },
    LAST30DAYS: {
      value: 'thirtyDays',
      description: 'The last 30 days.',
    },
  },
  description: 'An enum used to select information from the dmarc-report-api.',
})
